### ISSUE TITLE
[useResponseCache] Caching not working as expected when using federation + gateway

### ISSUE THREAD

**Issue by vinaybedre** [graphql-hive/envelop#2252]
### Issue workflow progress

_Progress of the issue based on the
[Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [x] 1. The issue provides a
     [minimal reproduction](https://en.wikipedia.org/wiki/Minimal_reproducible_example) available on
     [Stackblitz](https://stackblitz.com/fork/node).
  - _Please install the latest `@envelop/*` packages that you are using._
  - _Please make sure the reproduction is as small as possible._
- [ ] 2. A failing test has been provided
- [ ] 3. A local solution has been provided
- [ ] 4. A pull request is pending review
---

**Describe the bug**

The caching via plugin `useResponseCache` is not working as expected when used in the gateway of a federated service. The cache does not store ObjectTypes of GraphQL at all, but only the etag call.

**To Reproduce** Steps to reproduce the behavior:

1. Setup a federated service gateway.
2. Implement the `useResponseCache` plugin.
3. Perform GraphQL queries that should be cached.
4. Observe that only the etag calls are cached, but ObjectTypes are not.

**Expected behavior**

The cache should store ObjectTypes of GraphQL responses, ensuring that repeated queries retrieve data from the cache as expected.

**Environment:**

- OS: macOS
- NodeJS: v20.12.2
- `@envelop/*` versions:
    "@envelop/response-cache": "^6.1.2"

**Additional context**

A reproducible repo is provided here: [Stackblitz Reproduction](https://stackblitz.com/~/github.com/vinaybedre/envelop-response-cache-bug)

**Comment by EmrysMyrddin** [comment]
Hi! Thank you for raising this issue!

The problem seems to be the federation driver, which uses `useApolloFederation` plugin under the hood to setup a federation executor.

The problem is that this plugin is added at the end of the plugin list, which overrides the custom executor from the `useResponseCache`.

Do you have any warnings coming from the response cache plugin ? It should detect this case and warn about this problem.

To fix this, the federation yoga server should be updated to put the federation plugin at the start of the plugin list instead of the end.

**Comment by vinaybedre** [comment]
@EmrysMyrddin  you are correct, I am getting a warning `[useResponseCache] The cached execute function was not called, another plugin might have overwritten it. Please check your plugin order.`
But in federation, the plugin `useApolloFederation` is applied at the end as per this line https://github.com/dotansimha/graphql-yoga/blob/3d0cfd29fff92a98308b20b08c45aff68e0e9fa1/packages/nestjs-federation/src/index.ts#L111

or you mean to say that `useApolloFederation` must be at the start?

**Comment by vinaybedre** [comment]
@EmrysMyrddin I just created a patch by changing useApolloFederation to be the first plugin. You can see the changes at https://stackblitz.com/~/github.com/vinaybedre/envelop-response-cache-bug

After patching i see no warning, but caching is still not working.

![image](https://github.com/n1ru4l/envelop/assets/2630384/3e70e696-552c-4cab-b83c-22ae68ada1bd)
